exampleCourse/questions/element/matching/question.html

```diff
@@ -0,0 +1,154 @@
+
+<div class="card my-2">
+    <div class="card-body">
+        <pl-question-panel>
+            <p>
+                Example 4: if <code>number-options</code> is smaller than the amount of options
+                required to match the sampled statements, None of the Above will be added.
+            </p>
+        </pl-question-panel>
+            
+        <pl-matching answers-name="ex4" number-statements="3" number-options="2">
+            <pl-statement match="Ottawa">Canada</pl-statement>
+            <pl-statement match="Washington, D.C.">United States</pl-statement>
+            <pl-statement match="Mexico City">Mexico</pl-statement>
+            <pl-statement match="Paris">France</pl-statement>
+            
+            <pl-option>Tijuana</pl-option>
+            <pl-option>Toronto</pl-option>
+            <pl-option>New York City</pl-option>
+        </pl-matching>
+    </div>
+</div>
+
+<div class="card my-2">
+    <div class="card-body">
+        <pl-question-panel>
+            <p>
+                Example 5: the <code>counter-type</code> attribute indicates how options are identified. It can be set to <code>lower-alpha</code> (a, b, c, ...), <code>upper-alpha</code> (A, B, C, ...), or <code>decimal</code> (1, 2, 3).
+            </p>
+        </pl-question-panel>
+            
+        <pl-matching answers-name="ex5" counter-type="decimal" number-statements="3" number-options="3">
+            <pl-statement match="Ottawa">Canada</pl-statement>
+            <pl-statement match="Washington, D.C.">United States</pl-statement>
+            <pl-statement match="Mexico City">Mexico</pl-statement>
+            <pl-statement match="Paris">France</pl-statement>
+            
+            <pl-option>Tijuana</pl-option>
+            <pl-option>Toronto</pl-option>
+            <pl-option>New York City</pl-option>
+        </pl-matching>
+    </div>
+</div>
+
+<div class="card my-2">
+    <div class="card-body">
+        <pl-question-panel>
+            <p>
+                Example 6: options can be added manually if they require more than simple text.
+                All PrairieLearn and HTML elements are supported in options and statements.
+            </p>
+        </pl-question-panel>
+            
+        <pl-matching answers-name="ex6">
+            <pl-statement match="Matt">Matt</pl-statement>
+            <pl-statement match="Craig">Craig</pl-statement>
+            
+            <pl-option name="Matt"><pl-figure file-name="student_images/001.png" width="140px" directory="clientFilesCourse"></pl-figure></pl-option>
+            <pl-option name="Craig"><pl-figure file-name="student_images/002.png" width="140px" directory="clientFilesCourse"></pl-figure></pl-option>
+        </pl-matching>
+    </div>
+</div>
+
+
+<div class="card my-2">
+    <div class="card-body">
+        <pl-question-panel>
+            <p>
+                Example 7: Setting <code>counter-type="full-text"</code> will hide the column of options. The dropdown list
```

Comment:
This example 7 using `counter-type="full-text"` is great! I find it much less error prone than using letters as keys to match between statements and options.

I'm wondering why this isn't the default mode? From what I can see the only downside is if the options can't be displayed in a dropdown (e.g., equations, images, other complex HTML). If we have text options then I think I'd always want to use `full-text`.

As I was looking at this element, I was thinking to myself that I didn't like the matching via letter keys, and I was wishing that I could just directly match options to statements. For example, I was thinking about something like `<pl-order-blocks>` where the student could drag options next to matching statements. But then I saw Example 7 and I was amazed! This was exactly what I wanted! So my only question is why it was hidden as the 7th option rather than being the primary mode of use? 😄

Reply:
My initial design was a straightforward reimplementation of my LMS's matching quiz element, which uses numeric counters; full-text counters weren't even on my radar until Jonatan's suggestion. 

I personally feel that full-text is quite restrictive -- only being applicable for very short option texts, which in my personal use history is somewhat uncommon. But I am happy to make that the default and/or rearrange the examples if you want to promote that style as the default.

Reply:
I agree that it's probably fine to leave the default as is. But maybe we could elevate Example 7 up to Example 2?

Reply:
Done!

Reply:
Awesome, thanks!